### ISSUE TITLE
A few issues

### ISSUE THREAD

**Issue by PerJensen** [RiverVanRain/market#11]
Great work you've done with this plugin, and with upgrading to 3.0 @rohit1290. Thanks.

After testing I found a few issues though.

1. Language string hard-coded: https://github.com/RiverVanRain/market/blob/865f7f8c211e0313e7256624375719ae46c0430f/views/default/object/market/profile.php#L83

2. There are problems with image upload and deletion. When a new post is created without attaching an image, everything works fine, but if an image is attached, the page freezes when the save is clicked and this error message appears: HTTP ERROR 500.

However, the post is created and the image is stored in the data folder, but when viewing the post the picture is not displayed. If the post is deleted, the image remains in the data folder, it is not deleted.

3. The custom choices that can be defined under settings do not work when viewing the post (with custom select enabled). The string is not retrieved from the language file.

**Comment by rohit1290** [comment]
> Language string hard-coded:

This can be worked on. I will update it.

> There are problems with image upload and deletion.

What is is your elgg version and market plugin version? I have the plugin working without any issue with Elgg 3.0.1.

> The custom choices that can be defined under settings do not work when viewing the post (with custom select enabled). The string is not retrieved from the language file.

Will look into it.. give me some time.

**Comment by PerJensen** [comment]
Elgg 3.0.2, Market 3.0, tested with xampp latest.

**Comment by RiverVanRain** [comment]
> There are problems with image upload and deletion.

Confirmed.

**Comment by rohit1290** [comment]
@PerJensen 
> The string is not retrieved from the language file.

The language file is working fine. Did you flushed your cache and run and upgrade?

**Comment by PerJensen** [comment]
Yes, I'm pretty sure I've tested thoroughly. The only place I experience problems is with the language strings created by Custom Select Choices, such as: 'market:custom:unused' => "Unused".

I use a different language file than English, but as you can see from the attached image, the string is not even retrieved from the file en.php.

![market](https://user-images.githubusercontent.com/1220345/56470859-79c83700-644b-11e9-9e05-391b975c0417.jpg)

**Comment by rohit1290** [comment]
Ok, that place. found the problem.. working on it. (Screenshot always help). I thought that you were referring to the edit form's dropdown menu.

**Comment by rohit1290** [comment]
Regarding file upload and delete. I am not getting any error. Here's a screen-cast.

![Screencast 2019-04-21 18_52_46](https://user-images.githubusercontent.com/2772844/56470890-58764380-6469-11e9-83f2-b04989e31275.gif)

**Comment by RiverVanRain** [comment]
@PerJensen Have you installed Elgg using CLI (elgg-cli)?

**Comment by PerJensen** [comment]
Hmm, now we might get closer. Try with an image larger than 2000 or 3000px, More testing shows that I can use images smaler than that without problems. 

Besides, images stays in data folder, even if entire post is deleted.

**Comment by PerJensen** [comment]
The image upload issue seems to be with xampp. I get the same error when using File upload. I can't test it on a live server today, but will get back to you soon.

**Comment by RiverVanRain** [comment]
There's a bug with the rights permissions when running CLI https://github.com/Elgg/Elgg/issues/12545 
My reply in this discussion: https://elgg.org/discussion/view/2921089/installation-help-failed-to-create-cache-path#elgg-object-2921152
Keep in your mind this issue when you'll try to install the latest elgg on the server.

**Comment by PerJensen** [comment]
I'm not using CLI, but thanks.

**Comment by rohit1290** [comment]
@PerJensen 
> Hmm, now we might get closer. Try with an image larger than 2000 or 3000px, More testing shows that I can use images smaler than that without problems.

The issue is not with Market plugin but with Elgg. I tried uploading 2000+ px image using file plugin and it showed an error "Form is missing __token or __ts fields". I am getting the same error when uploading the same image file in market plugin.

Can you please create a ticket on Elgg repo for this issue.

For now all other issue has been addressed in https://github.com/RiverVanRain/market/pull/12 and https://github.com/RiverVanRain/market/pull/13

**Comment by PerJensen** [comment]
#12 and #13  tested on xampp and both fix are working. Great.
About the upload issue: want to check it out on my live server tomorrow. Will keep you posted.

**Comment by PerJensen** [comment]
Now I have checked the upload of images on my server and it works with both the Files and Market plugin. I have tried with various images of about 4600 x 3400px - 3.14MB without any problems.

When I delete the post, the pictures are also deleted from the data folder.

However, these tests of the Market plugin have revealed another error both in xampp and on live server, which has nothing to do with Market. I cannot change the CSS in Market in the recommended manner. When I overwrite the CSS file by placing market CSS in my own plugin containing all my changes, it has no effect.

Maybe it's because I can't change the order of plugins in the plugin list in the administration. In earlier versions of Elgg, it was crucial that the plugin containing your changes was at the bottom of the list. Don't know if it's still required.

I do not intend to create a ticket on Elgg repo for this issue, since I really no longer work with github (or Elgg).

But again, thank you for all your work with Market and for checking and correcting these errors.
